crates/derive/src/pipeline/builder.rs

```diff
@@ -32,9 +32,8 @@ where
     dap_source: Option<D>,
     chain_provider: Option<P>,
     builder: Option<B>,
+    origin: Option<BlockInfo>,
```

Comment:
hmm - I don't think we should ever be setting the L1 origin of the pipeline manually, yeah? We should initialize it with an L2 safe head, and find the L1 origin through the L2 chain provider?

Reply:
This is just a matter of prefetching the data. I'd prefer to keep the pipeline construction synchronous and not embed this async fetching of the associated l1 origin.